File: server/src/config/momo.js
export const momoConfig = {
  env: process.env.MOMO_ENV || 'mock',
  partnerCode: process.env.MOMO_PARTNER_CODE || '',
  accessKey: process.env.MOMO_ACCESS_KEY || '',
  secretKey: process.env.MOMO_SECRET_KEY || '',
  endpoint: process.env.MOMO_ENDPOINT || 'https://test-payment.momo.vn/v2/gateway/api/create',
  queryEndpoint: process.env.MOMO_QUERY_ENDPOINT || 'https://test-payment.momo.vn/v2/gateway/api/query',
  redirectUrl: process.env.MOMO_REDIRECT_URL || 'http://localhost:5000/api/payments/momo/return',
  ipnUrl: process.env.MOMO_IPN_URL || process.env.MOMO_NOTIFY_URL || 'http://localhost:5000/api/payments/momo/ipn',
  requestType: process.env.MOMO_REQUEST_TYPE || 'captureWallet',
  clientUrl: process.env.CLIENT_URL || 'http://localhost:5173'
};

export const isMomoMock = () => momoConfig.env === 'mock';

File: server/src/config/momo.py
from src.config.settings import settings

momo_config = {
    "env": settings.MOMO_ENV,
    "partnerCode": settings.MOMO_PARTNER_CODE,
    "accessKey": settings.MOMO_ACCESS_KEY,
    "secretKey": settings.MOMO_SECRET_KEY,
    "endpoint": settings.MOMO_ENDPOINT,
    "queryEndpoint": settings.MOMO_QUERY_ENDPOINT,
    "redirectUrl": settings.MOMO_REDIRECT_URL,
    "ipnUrl": settings.MOMO_IPN_URL or settings.MOMO_NOTIFY_URL,
    "requestType": settings.MOMO_REQUEST_TYPE,
    "clientUrl": settings.CLIENT_URL,
}

def is_momo_mock() -> bool:
    return momo_config["env"] == "mock"
